sapio-trait/src/lib.rs

```diff
@@ -1,17 +1,16 @@
-use schemars::*;
+use sapio_data_repr::{SapioModuleBoundaryRepr, SapioModuleSchema};
 use serde::*;
-use serde_json::Value;
 
 pub trait SapioAPIHandle {
-    fn get_api(&self) -> serde_json::Value;
+    fn get_api(&self) -> SapioModuleSchema;
 }
-impl SapioAPIHandle for serde_json::Value {
+impl SapioAPIHandle for SapioModuleSchema {
     fn get_api(&self) -> Self {
         self.clone()
     }
 }
-pub trait SapioJSONTrait: JsonSchema + Serialize + for<'a> Deserialize<'a> {
-    fn get_example_for_api_checking() -> Value;
+pub trait SapioSchemaValidatable: Serialize + for<'a> Deserialize<'a> {
+    fn get_example_for_api_checking() -> SapioModuleBoundaryRepr;
```

Comment:
Can this be a Self if we have a bound that Self can be converted to a  SapioModuleBoundryRepr